File: src/main/java/org/jukeboxmc/world/generator/Generator.java
package org.jukeboxmc.world.generator;

import org.jukeboxmc.math.Vector;
import org.jukeboxmc.world.chunk.Chunk;
import org.jukeboxmc.world.chunk.manager.PopulationChunkManager;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public abstract class Generator {

    public abstract void generate( Chunk chunk, int chunkX, int chunkZ );

    public abstract void populate( PopulationChunkManager manager, int chunkX, int chunkZ );

    public abstract void finish( PopulationChunkManager manager, int chunkX, int chunkZ );

    public abstract Vector getSpawnLocation();
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/world/generator/Generator.kt
package org.jukeboxmc.api.world.generator

import org.jukeboxmc.api.math.Vector
import org.jukeboxmc.api.world.chunk.Chunk


abstract class Generator {
    abstract fun generate(chunk: Chunk, chunkX: Int, chunkZ: Int)
    abstract fun populate(manager: PopulationChunkManager, chunkX: Int, chunkZ: Int)
    abstract fun finish(manager: PopulationChunkManager, chunkX: Int, chunkZ: Int)
    abstract fun spawnLocation(): Vector
}

